engines/director/lingo/lingo-codegen.cpp

```diff
@@ -527,6 +527,8 @@ void Lingo::varAssign(Datum &var, Datum &value) {
 			sym->u.s = value.u.s;
 		} else if (value.type == VOID) {
 			sym->u.i = 0;
+		} else if (value.type == ARRAY) {
+			sym->u.farr = new DatumArray(*value.u.farr);
```

Comment:
are you sure we need to copy the whole thing and not just pass the reference here?

Reply:
Pass by reference works.